File: public/cses/sorting-and-searching/concert-tickets/lb_approach/concert-tickets.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // setup
    // setup-start
#ifndef ONLINE_JUDGE
    freopen("../test_cases/1/input.txt", "r", stdin);
    freopen("../test_cases/1/output.txt", "w", stdout);
#endif
    // setup-end

    int n, m;
    cin >> n >> m;
    multiset<int> tickets;
    vector<int> bids(m);
    for (int i = 0; i < n; i++) {
        int v;
        cin >> v;
        tickets.insert(v);
    }
    for (int i = 0; i < m; i++) {
        cin >> bids[i];
    }
    for (int bid : bids) {
        auto it = tickets.lower_bound(bid);
        if (it != tickets.end() && *it == bid) {
            cout << *it << endl;
            tickets.erase(it);
        } else {
            // we have a smaller value
            if (it != tickets.begin()) {
                it--;
                cout << *it << endl;
                tickets.erase(it);
            } else {
                cout << -1 << endl;
            }
        }
    }
}

File: public/cses/sorting-and-searching/concert-tickets/lb_approach/concert-tickets.py
# setup-start
import os
import sys
script_directory = os.path.dirname(os.path.realpath(__file__))
parent_directory = os.path.dirname(script_directory)
input_path = os.path.join(parent_directory, 'test_cases/1/input.txt')
output_path = os.path.join(parent_directory, 'test_cases/1/output.txt')
if os.path.exists(input_path):
    sys.stdin = open(input_path, 'r')
    try:
        sys.stdout = open(output_path, 'w')
    except Exception as e:
        pass
# setup-end


def main():
    import bisect

    # Read inputs
    n, m = map(int, input().split())
    tickets = sorted(map(int, input().split()))
    bids = list(map(int, input().split()))

    for bid in bids:
        # Find first element >= bid
        idx = bisect.bisect_left(tickets, bid)

        if idx < len(tickets) and tickets[idx] == bid:
            # Exact match
            print(tickets[idx])
            tickets.pop(idx)
        else:
            # Find smaller value if available
            if idx == 0:
                print(-1)
            else:
                print(tickets[idx - 1])
                tickets.pop(idx - 1)


main()
